core/sqf/sql/scripts/get_libhdfs_files

```
@@ -0,0 +1,186 @@
+HADOOP_ID=hadoop-2.6.0
+HADOOP_SRC_ID=${HADOOP_ID}-src
+HADOOP_SRC_TAR=${HADOOP_SRC_ID}.tar.gz
+
+# files to build required version of Google Protocol Buffers
+PROTOBUF_MIRROR_URL=https://github.com/google/protobuf/releases/download/v2.5.0
+PROTOBUF_ID=protobuf-2.5.0
+PROTOBUF_TAR=${PROTOBUF_ID}.tar.gz
+
+# result of protobuf build
+PROTOBUF_TGT_ID=protobuf-tgt
+
+# Directories to copy the built libhdfs library and corresponding include file
+TGT_INC_DIR=$MY_SQROOT/export/include
+TGT_LIB_DIR=$MY_SQROOT/export/lib${SQ_MBTYPE}
+
+FORCE_BUILD=false
+VERBOSE=false
+
+USAGE="Usage $0 [ -f | --force ]\
+                [ -v | --verbose ]\
+                [ -d <temp dir> | --tempDir <temp dir> ]"
+
+while [[ $# > 0 ]]
+do
+  arg="$1"
+
+  case $arg in
+    -f|--force)
+       FORCE_BUILD=true
+       ;;
+
+    -v|--verbose)
+       VERBOSE=true
+       ;;
+
+    -d|--tempDir)
+       shift
+       if [[ $# < 1 ]]; then
+         echo "Expecting argument after -d or --tempDir"
+         exit 1
+       fi
+       LIBHDFS_TEMP_DIR="$1"
+       ;;
+
+    *)
+       echo "Unknown command line option: $arg"
+       echo "Usage $0 [ -f | --force ]"
+       exit 1
+    ;;
+  esac
+
+  shift
+done
+
+
+if [[ $FORCE_BUILD == true || \
+      ! -e ${TGT_INC_DIR}/hdfs.h || \
+      ! -e ${TGT_LIB_DIR}/libhdfs.so ]]; then
+
+  if [ ! -d $LIBHDFS_TEMP_DIR ]; then
+    mkdir $LIBHDFS_TEMP_DIR
+  fi
+
+  cd $LIBHDFS_TEMP_DIR
+
+  if [[ ! -f ${PROTOBUF_TAR} ]]; then
+    echo "Downloading Google Protocol Buffers..." | tee -a ${LOGFILE}
+    wget ${PROTOBUF_MIRROR_URL}/${PROTOBUF_TAR} >${LOGFILE}
+  fi
+
+  if [[ $FORCE_BUILD == true ]]; then
+    rm -rf ${LIBHDFS_TEMP_DIR}/${PROTOBUF_ID}
+    rm -rf ${LIBHDFS_TEMP_DIR}/${PROTOBUF_TGT_ID}
+  fi
+
+  if [[ ! -d ${PROTOBUF_ID} ]]; then
+    echo "Unpacking Google Protocol Buffer tar file..." | tee -a ${LOGFILE}
+    rm -rf ${LIBHDFS_TEMP_DIR}/${PROTOBUF_TGT_ID}
```

The "-rf" option will suppress errors if the files don't exist.